azurerm/validators_test.go

```diff
@@ -128,3 +128,42 @@ func TestValidateIso8601Duration(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateAzureTimeZone(t *testing.T) {
```

Comment:
_minor_ I would question the need for this validation as we are just doing a "is the string in this array" check using the built in validator.